src/main/java/net/earthcomputer/clientcommands/command/CrackVillagerRNGCommand.java

```diff
@@ -2,42 +2,85 @@
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher, CommandBuildContext context) {
         dispatcher.register(literal("ccrackvillager")
-                .then(literal("cancel")
-                        .executes(ctx -> cancel(ctx.getSource())))
-                .then(literal("clock")
-                        .then(argument("clockpos", blockPos())
-                                .executes(ctx -> crackVillagerRNG(ctx.getSource(), getBlockPos(ctx, "clockpos"))))
-                                )
-                .then(literal("interval")
-                        .then(argument("ticks", integer(0, 10))
-                                .executes(ctx -> setInterval(ctx.getSource(), getInteger(ctx, "ticks")))))
-                .then(literal("enchant").then(argument("name", itemAndEnchantmentsPredicate().withItemPredicate((i) -> i.equals(Items.BOOK)).constrainMaxLevel())
-                        .executes(ctx -> lookingForEnchantment(ctx.getSource(), getItemAndEnchantmentsPredicate(ctx, "name"))))));
+            .then(literal("cancel")
+                .executes(ctx -> cancel(ctx.getSource())))
+            .then(literal("clock")
+                .then(argument("clockpos", blockPos())
+                    .executes(ctx -> crackVillagerRNG(ctx.getSource(), getBlockPos(ctx, "clockpos")))))
+            .then(literal("interval")
+                .then(argument("ticks", IntegerArgumentType.integer(0, 20))
+                    .executes(ctx -> setInterval(ctx.getSource(), getInteger(ctx, "ticks")))))
+            .then(literal("addgoal")
+                .then(genFirst(context))
+                .then(genSecond(context))
+                .then(genResult(context)))
+            .then(literal("listgoals")
+                .executes(CrackVillagerRNGCommand::listGoals))
+            .then(literal("removegoal")
+                .then(argument("index", integer(1,CCrackVillager.goalOffers::size))
+                    .executes(CrackVillagerRNGCommand::removeGoal)))
+            .then(literal("cleargoals").executes(CrackVillagerRNGCommand::clearGoals))
+            .then(literal("run").executes(CrackVillagerRNGCommand::doRun)));
+    }
+
+    private static int doRun(CommandContext<FabricClientCommandSource> context) {
+        CCrackVillager.findingOffers = true;
+        if(CCrackVillager.goalOffers.isEmpty()) {
+            context.getSource().sendFeedback(Component.translatable("commands.ccrackvillager.emptyGoals"));
+        }
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int clearGoals(CommandContext<FabricClientCommandSource> context) {
+        CCrackVillager.goalOffers.clear();
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int removeGoal(CommandContext<FabricClientCommandSource> context) {
+        context.getSource().sendFeedback(Component.literal("remove goal: " + CCrackVillager.goalOffers.remove(getInteger(context, "index")-1)));
```

Comment:
reformat to add spacing

```suggestion
        context.getSource().sendFeedback(Component.literal("remove goal: " + CCrackVillager.goalOffers.remove(getInteger(context, "index") - 1)));
```